Skip the long path tests when not on Windows

rsp opened the pull request:
This is another attempt to fix the tests failing on Ubuntu (see issue #2255)
This time by skipping the tests when not on Windows (as [advised](https://github.com/nodejs/node/pull/3929#issuecomment-161072622) by @bnoordhuis)

Other PRs that solve the same issue:
* #3925 runs the tests under os.tmpdir() on Linux if it is readable/writable (my first idea)
* #3929 retries the tests on Linux under os.tmpdir() only when they fail (as [advised](https://github.com/nodejs/node/pull/3925#issuecomment-158204979) by @rvagg)
* #4116 (this one) skips the test when not on Windows (as [advised](https://github.com/nodejs/node/pull/3929#issuecomment-161072622) by @bnoordhuis)

See also comments to PRs #3925 and #3929 for more context.

bnoordhuis commented on test/parallel/test-fs-long-path.js:
```diff
@@ -1,4 +1,7 @@
 'use strict';
+
+if (process.platform !== 'win32') process.exit();
```
Can you use the TAP-compatible approach we use e.g. [here](https://github.com/nodejs/node/blob/v5.1.0/test/parallel/test-spawn-cmd-named-pipe.js#L6-L9)?

(I forgot about `common.isWindows`, it's fairly new.)

rsp replied:
I changed the test skipping in 5502b9b - is that what you mean?

bnoordhuis commented on test/parallel/test-fs-long-path.js:
```diff
@@ -4,6 +4,11 @@ var fs = require('fs');
 
+if (!common.isWindows) {
+  console.log('1..0 # Skipped: this test is Windows-specific.');
+  process.exit();
```
Can you use `return` here?  That's a little more idiomatic, that's what other tests do.  LGTM apart from that.

rsp replied:
@bnoordhuis Ok, done - in 09691de. Please let me know if I should squash the commits now or still wait for more feedback. Thanks.

bnoordhuis commented:
LGTM, thanks.  Can you squash and write a commit log in the style that CONTRIBUTING.md wants?

CI: https://ci.nodejs.org/job/node-test-pull-request/904/

rsp commented:
@bnoordhuis I rebased it on fresh master, squashed into e07b145 with new message and force pushed here. I can't access the CI link ("rsp is missing the Overall/Read permission") so I don't know if it tests the forced update automatically or not. Thanks for your help.

jasnell commented:
LGTM

JungMinu commented:
LGTM

JungMinu commented:
As @rsp updated, new CI : https://ci.nodejs.org/job/node-test-pull-request/932/

JungMinu commented:
LGTM if new CI is happy :)

bnoordhuis commented:
Thanks, landed in 19e06d7.

rsp commented:
@jasnell @JungMinu @bnoordhuis @rvagg Thanks. My pleasure. I think the `wontfix` label can be removed from issue #2255 now.
